### PR TITLE
Hacktoberfest

### PR DESCRIPTION
**Pull request by HarryET**
Hey, here is my first go at making the changes

### CODE REVIEW

**Comment by HarryET** [comment]
Updated the repo after watching video 
https://www.youtube.com/watch?v=0405xzlycD8

File: tox.ini [hunk 1]
```diff
@@ -3,8 +3,8 @@ envlist =
     py36,
     py37,
     py38,
-    flake8,
-    coverage
+    coverage,
+    pre-commit
```
Comment (jugmac00) [inline tox.ini:8]: Right! That's correct. `flake8` does not need to get run any more, but we now want the `pre-commit` env to run.

Nothing to do here.

Comment (HarryET) [reply tox.ini:8]: ok, when should that run?

File: tox.ini [hunk 2]
```diff
@@ -22,18 +22,22 @@ commands =
...
-    flake8-bugbear
-commands =
-    flake8 src tests setup.py
+[testenv:flake8]	
+deps =	
+    flake8	
+    flake8-click	
+    flake8-bugbear	
+commands =	
+    flake8 src tests setup.py
```
Comment (jugmac00) [inline tox.ini:31]: We do not need the `flake8` env any more.

But please do not forget to add both `flake8-click` and `flake8-bugbear` as additional dependencies to the flake8 configuration block in the pre-commit-config.yaml file.

You can compare with https://github.com/jugmac00/flask-reuploaded/blob/master/.pre-commit-config.yaml

File: tox.ini [hunk 3]
```diff
@@ -22,18 +22,22 @@ commands =
...
+    flake8-click	
+    flake8-bugbear	
+commands =	
+    flake8 src tests setup.py	
 
-[flake8]
-max-line-length = 88
+[flake8]	
+max-line-length = 88	
 ignore = E231  # clashes with black
```
Comment (jugmac00) [inline tox.ini:35]: Please remove the trailing white spaces behind [flake8] and 88.

File: tox.ini (hunk 2 above)
Comment (HarryET) [reply tox.ini:31]: ok so remve from the .ini and add the deps to .yaml

Comment (HarryET) [reply tox.ini:31]: remove this?
```yaml
[testenv:flake8]	
deps =	
    flake8	
    flake8-click	
    flake8-bugbear	
commands =	
    flake8 src tests setup.py	

[flake8]	
max-line-length = 88	
ignore = E231  # clashes with black
```

File: tox.ini [hunk 4]
```diff
@@ -22,18 +22,22 @@ commands =
...
 
+[testenv:pre-commit]
+skip_install = true
+deps = pre-commit
+commands = pre-commit run --all-files --show-diff-on-faliure
```
Comment (jugmac00) [inline tox.ini:41]: This looks almost good - but this also tells me you did not run it :-)

There is a typo in failure which results in a `pre-commit: error: unrecognized arguments: --show-diff-on-faliure`

Ideally, you should have `tox` installed on your system to be able to execute both the tests and the linter.

If you are interested in how to setup `tox`, you could have a look at this long issue https://github.com/morepath/morepath_cerebral_todomvc/issues/2 or go to `tox`s documentation:
https://tox.readthedocs.io/en/latest/install.html

I can support you - but that is up to you :-)

**Review by jugmac00 — CHANGES_REQUESTED** [review]
We are on a good way - please have a look at the suggested changes!

Thank you!

File: tox.ini (hunk 4 above)
Comment (HarryET) [reply tox.ini:41]: Fixed the type!

File: tox.ini (hunk 3 above)
Comment (HarryET) [reply tox.ini:35]: removed whiteshpace

File: tox.ini (hunk 1 above)
Comment (jugmac00) [reply tox.ini:8]: All the envs listed in the envlist block are run, when you just type `tox` (after you installed it).

You could select a single env to run via e.g. `tox -e pre-commit` and then only the pre-commit env is running.

File: tox.ini (hunk 2 above)
Comment (jugmac00) [reply tox.ini:31]: You should remove the block with [testenv:flake8] as this configures the env which is now run via pre-commit.

But you should not remove the [flake8] block, as this is just a general configuration for flake8, not matter which tool runs flake8.

You can put configuration in several files in Python, e.g. in a .flake8 or a tox.ini or a setup.cfg or nowadays also in pyproject.toml.

I just keep all my configuration in tox.ini - thats what I am used to and thats what I prefer for my projects.

**Comment by HarryET** [comment]
After those commits whats left now? apart from changes to maybe this area? https://github.com/HarryET/hibpcli/blob/master/tox.ini#L25

**Comment by jugmac00** [comment]
> After those commits whats left now? apart from changes to maybe this area? https://github.com/HarryET/hibpcli/blob/master/tox.ini#L25

That's pretty much it! Delete the now obsolete env for flake8 (but not its configuration) and we are good to go.

**Comment by HarryET** [comment]
done!

**Comment by jugmac00** [comment]
Thanks! LGTM! 1/4 of a super cool shirt well deserved :-)

**Pull request merged**